### PR TITLE
feat(codecatalyst-issues): adding issue support

### PR DESCRIPTION
**Pull request by tramo-usg**
### Description

- Add issue component, see https://docs.aws.amazon.com/codecatalyst/latest/userguide/issues.html for more documentation
- Add issue deduplication in resynthesis
- Add issue examples to test blueprint

### Testing

local build, testing yarn blueprint:synth and yarn blueprint:resynth on test blueprint

### Additional context

Merge pending integration and testing with blueprint deployment system 

### Example 1: Creating an issue

```
import { Issue } from '@amazon-codecatalyst/blueprint-component.issues';
...
new Issue(this, 'myFirstIssue', {
  title: 'myFirstIssue',
  content: 'this is an example issue',
});
```

### Example 2: Creating a high priority issue

```
import { Issue } from '@amazon-codecatalyst/blueprint-component.issues';
...
new Issue(this, 'mySecondIssue', {
  title: 'mySecondIssue',
  content: 'this is an example high priority issue',
  priority: 'HIGH',
});
```

### Example 3: Creating a low priority issue with labels

```
import { Issue } from '@amazon-codecatalyst/blueprint-component.issues';
...
new Issue(this, 'myThirdIssue', {
  title: 'myThirdIssue',
  content: 'this is an example of a low priority issue with a label',
  priority: 'LOW',
  labels: ['exampleLabel'],
});
```

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by alexforsyth** [comment]
Can you update the description to include example usage of the issues component?

File: packages/blueprints/blueprint/src/blueprint.ts [hunk 1]
```diff
@@ -142,12 +142,18 @@ export class Blueprint extends Project {
...
+    const supersetNonSourceNonIssuesPaths: string[] = filepathSet([proposedBundle], ['**/*', '!src/**', '!issues/**']);
+    for (const filepath of supersetNonSourceNonIssuesPaths) {
+      this.writeNonSourceFile(filepath, proposedBundle);
+    }
+
+    /**
+     * copy all issue src files that are unique to the proposedBundle into the resolved bundle
+     * to prevent duplicate issue creation
+     */
+    const setUniqueToProposedBundle = filepathDiffBetweenSets_B_A(ancestorBundle, proposedBundle, ['issues/**']);
```
Comment (alexforsyth) [inline packages/blueprints/blueprint/src/blueprint.ts:154]: This diff should be handled by the issue component

File: packages/blueprints/blueprint/src/blueprint.ts [hunk 2]
```diff
@@ -142,12 +142,18 @@ export class Blueprint extends Project {
      * copy all non-src file from proposedBundle into the resolved bundle
      * only src is merge constructed.
      */
-    const supersetNonSourcePaths: string[] = filepathSet([proposedBundle], ['**/*', '!src/**']);
-    for (const filepath of supersetNonSourcePaths) {
-      const outputPath = path.join(this.outdir, filepath);
-      fs.mkdirSync(path.dirname(outputPath), { recursive: true });
-      const filecontent = fs.readFileSync(path.join(proposedBundle, filepath));
-      fs.writeFileSync(outputPath, filecontent);
+    const supersetNonSourceNonIssuesPaths: string[] = filepathSet([proposedBundle], ['**/*', '!src/**', '!issues/**']);
```
Comment (alexforsyth) [inline packages/blueprints/blueprint/src/blueprint.ts:145]: You dont need to update the base blueprint here

Comment (alexforsyth) [reply packages/blueprints/blueprint/src/blueprint.ts:145]: Just make sure that the proposed bunlde containts the rigt issues

**Review by alexforsyth — CHANGES_REQUESTED** [review]

File: packages/blueprints/blueprint/src/blueprint.ts (hunk 2 above)
Comment (tramo-usg) [reply packages/blueprints/blueprint/src/blueprint.ts:145]: resolved through discussion

File: packages/blueprints/blueprint/src/blueprint.ts (hunk 1 above)
Comment (tramo-usg) [reply packages/blueprints/blueprint/src/blueprint.ts:154]: resolved through discussion

**Review by alexforsyth — APPROVED** [review]
LGTM

**Pull request merged**